libvast/native-plugins/feather.cpp

```diff
@@ -159,8 +184,7 @@ class active_feather_store final : public active_store {
     // TODO: Set write_properties.chunksize to the expected batch size
     write_properties.compression = arrow::Compression::ZSTD;
     write_properties.compression_level
-      = arrow::util::Codec::DefaultCompressionLevel(arrow::Compression::ZSTD)
-          .ValueOrDie();
+      = detail::narrow<int>(feather_config_.zstd_compression_level);
```

Comment:
Do we need to validate here that the level is valid for the codec? Same for Parquet.

Reply:
I tested it with out-of-bounds numbers, and it seems to work / accept the nearest valid value.

However, verifying the config is a good idea, but at this particular stage, it's too late IMHO - we should produce a message on startup.

Reply:
The last unaddressed point - is there a place where the basic config is verified? I know there's a place where we "translate" legacy config bits...

Reply:
This is probably the correct spot for it.

https://github.com/tenzir/vast/blob/2edd66e81d3e60ea5ced22f2fd30271f54f031ad/vast/vast.cpp#L187-L204

Reply:
done in latest [commit](https://github.com/tenzir/vast/pull/2623/commits/dda026d3e15ee9a07cc4d5b5810b32f9051b16b6). It's a bit convoluted, and also the range according to `arrow::util::Codec::M{in,ax}imumCompressionLevel(arrow::Compression::ZSTD)` is `[-131072, 22]`, so the lower bound is not reliable.